lib/srv/discovery/discovery.go

```diff
@@ -198,3 +267,54 @@ func (s *Server) Wait() error {
 	}
 	return nil
 }
+
+func (s *Server) getAzureSubscriptions(ctx context.Context, subs []string) ([]string, error) {
+	subscriptionIds := subs
+	if slices.Contains(subs, types.Wildcard) {
+		subsClient, err := s.Clients.GetAzureSubscriptionClient()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		subscriptionIds, err = subsClient.ListSubscriptionIDs(ctx)
+		return subscriptionIds, trace.Wrap(err)
+	}
+
+	return subscriptionIds, nil
+}
+
+func (s *Server) initTeleportResourceWatchers(initNode bool) (err error) {
+	if !initNode {
+		return nil
+	}
+	s.nodeWatcher, err = services.NewNodeWatcher(s.ctx, services.NodeWatcherConfig{
+		ResourceWatcherConfig: services.ResourceWatcherConfig{
+			Component: teleport.ComponentDiscovery,
+			Log:       s.Log,
+			Client:    s.AccessPoint,
+		},
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// wait for nodeWatcher to complete initialization so the
+	// discovery server doesn't get an empty list of nodes and attempt
+	// to perform an installation on all the nodes that do actually
+	// exist.
+	t := time.NewTicker(500 * time.Millisecond)
+	defer t.Stop()
+	for initNode {
```

Comment:
As it stands, this will block the service startup until the node watcher has been initialized. I know this was introduced as a part of discovery service and I've created a ticket to address this in v10: https://github.com/gravitational/teleport/issues/17146.

But I think we should do this in a way that does not block the service startup. For example, can the goroutine that actually runs SSM installer block until nodeWatcher is initialized, not the entire service?

cc @lxea